.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,19 @@
+name: Publish Docs via GitHub Pages
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Deploy Docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@v1
+
+      - name: Deploy docs
+        uses: mhausenblas/mkdocs-deploy-gh-pages@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CUSTOM_DOMAIN: terrahab.github.io
```

Comment:
Has this workflow been tested successfully?

Reply:
I installed https://github.com/nektos/act to test it but I'm not sure if it's working. Gonna test some more

Reply:
Excellent! Once you're good with this, go ahead and rebase to squash these to a single commit and we can merge

Reply:
```
✦ ❯ act -l
 ╭─────────────╮
 │ Deploy Docs │
 ╰─────────────╯

TerraMICRO on  mkdocs 
✦ ❯ act
[Publish Docs via GitHub Pages/Deploy Docs] 🚀  Start image=node:12.6-buster-slim
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker run image=node:12.6-buster-slim entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker cp src=/home/phil/repos/terrahab/TerraMICRO/. dst=/github/workspace
[Publish Docs via GitHub Pages/Deploy Docs] ⭐  Run Checkout master
[Publish Docs via GitHub Pages/Deploy Docs]   ✅  Success - Checkout master
[Publish Docs via GitHub Pages/Deploy Docs] ⭐  Run Deploy docs
[Publish Docs via GitHub Pages/Deploy Docs]   ☁  git clone 'https://github.com/mhausenblas/mkdocs-deploy-gh-pages' # ref=master
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker build -t act-mhausenblas-mkdocs-deploy-gh-pages-master:latest /home/phil/.cache/act/mhausenblas-mkdocs-deploy-gh-pages@master
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker run image=act-mhausenblas-mkdocs-deploy-gh-pages-master:latest entrypoint=[] cmd=[]
| INFO: Setting custom domain for github pages
| nektos/act
| nektos/act@users.noreply.github.com
| error: refs/heads/master does not point to a valid object!
| error: refs/remotes/origin/HEAD does not point to a valid object!
| error: refs/remotes/origin/master does not point to a valid object!
| INFO    -  Cleaning site directory 
| INFO    -  Building documentation to directory: /github/workspace/site 
| INFO    -  Documentation built in 0.15 seconds 
| INFO    -  Copying '/github/workspace/site' to 'gh-pages' branch and pushing to GitHub. 
| ERROR   -  Failed to deploy to GitHub with error: 
| error: refs/heads/master does not point to a valid object!
| error: refs/remotes/origin/HEAD does not point to a valid object!
| error: refs/remotes/origin/master does not point to a valid object!
| error: refs/heads/master does not point to a valid object!
| error: refs/remotes/origin/HEAD does not point to a valid object!
| error: refs/remotes/origin/master does not point to a valid object!
| fatal: no path specified; see 'git help pull' for valid url syntax
|  
[Publish Docs via GitHub Pages/Deploy Docs]   ❌  Failure - Deploy docs
Error: exit with `FAILURE`: 1
```

I say we roll with it

Reply:
Can you squash without rebasing? I usually follow this flow: https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

Reply:
Either way, it's done now. I'll go ahead and merge it